src/shared/lib/table.ts

```diff
@@ -78,3 +79,67 @@ export function useTable<T>(fetchFn: TableFetchFn<T>) {
     setSize,
   };
 }
+// for generating blocksListTable in paginated way
+export function useBlockTable(selectedTransaction: TransactionDto[]) {
+  const loading = ref(false);
+  const items: Ref<Transaction[]> = ref([]);
+  const pagination = reactive<TablePagination>({
+    page: 1,
+    page_size: 10,
+    pages: 1,
+    total: 10,
+  });
+
+  async function fetch() {
+    loading.value = true;
+    const res = await transactionModel.fetchBlocksList(selectedTransaction, {
+      page: pagination.page,
+      page_size: pagination.page_size,
+    });
+    items.value = res.data;
+    pagination.page = res.pagination.page;
+    pagination.page_size = res.pagination.page_size;
+    pagination.total = res.pagination.total;
+    pagination.pages = Math.ceil(res.pagination.total / res.pagination.page_size);
```

Comment:
You can define `pagination` with `ref()` instead of `reactive` and replace the whole object at once:

```ts
const pagination = ref<null | TablePagination>(null)

// ...

pagination.value = {
  ...res.pagination,
  pages: Math.ceil(res.pagination.total / res.pagination.page_size)
}
```

Reply:
While I kept most of the code similar to the previous implementation. Using ```ref()``` gives more advantage. 
Acknowledged and committed 👍🏼